lib/multigrid.cpp

```diff
@@ -347,6 +347,14 @@ namespace quda
 
     param_presmooth->Nkrylov = param_presmooth->maxiter;
     param_presmooth->pipeline = param_presmooth->maxiter;
+
+    if (param_presmooth->inv_type == QUDA_CA_GCR_INVERTER || param_presmooth->inv_type == QUDA_CA_CG_INVERTER ||
```

Comment:
Would an `is_ca_solver` method be helpful?
```c++
bool is_ca_solver(QudaInvertType type)
{
  switch (type) {
  case QUDA_CA_GCR_INVERTER:
  case QUDA_CA_CG_INVERTER:
  case QUDA_CA_CGNR_INVERTER:
  case QUDA_CA_CGNE_INVERTER:
    return true;   
  default: return false  
}
```

Reply:
Haha, yes, it would, I'll slot that in somewhere. Sometimes I need to do something extra for CA-GCR but that becomes a neater additional conditional.

Reply:
https://github.com/lattice/quda/commit/06c45e4fa09f664cfc83aca932d79719bf6f65c5